pyomo/contrib/parmest/utils/model_utils.py

```diff
@@ -43,21 +43,25 @@ def convert_params_to_vars(model, param_names=None, fix_vars=False):
 
     model = model.clone()
 
-    if param_names is None:
-        param_names = [param.name for param in model.component_data_objects(pyo.Param)]
+    if param_CUIDs is None:
+        param_CUIDs = [
+            ComponentUID(param) for param in model.component_data_objects(pyo.Param)
+        ]
 
-    indexed_param_names = []
+    # keep a list of the parameter CUIDs in the case of indexing
+    indexed_param_CUIDs = []
 
     # Convert Params to Vars, unfix Vars, and create a substitution map
     substitution_map = {}
     comp_map = ComponentMap()
-    for i, param_name in enumerate(param_names):
+    for param_CUID in param_CUIDs:
+
         # Leverage the parser in ComponentUID to locate the component.
-        theta_cuid = ComponentUID(param_name)
-        theta_object = theta_cuid.find_component_on(model)
+        theta_object = param_CUID.find_component_on(model)
 
         # Param
         if theta_object.is_parameter_type():
+
             # Delete Param, add Var
             vals = theta_object.extract_values()
             model.del_component(theta_object)
```

Comment:
I don't think this will work on hierarchical models. I think we need to add more tests for edge cases.

Reply:
I think I fixed this and added a test.